app/routes/challenges/get-preview.js

```diff
@@ -3,7 +3,10 @@ import Ember from 'ember';
 export default Ember.Route.extend({
 
   model(params) {
-    return this.get('store').findRecord('challenge', params.challenge_id);
+    return this
+      .get('store')
+      .findRecord('challenge', params.challenge_id)
+      .then((challenge) => challenge);
```

Comment:
je pense qu'il n'y a pas besoin du `then` et que retourner la promesse peut suffire.



> This hook follows the asynchronous/promise semantics described in the documentation for beforeModel
> – http://emberjs.com/api/classes/Ember.Route.html#method_model

> You can return a promise from this hook to pause the transition until the promise resolves (or rejects). This could be useful, for instance, for retrieving async code from the server that is required to enter a route.
> – http://emberjs.com/api/classes/Ember.Route.html#method_beforeModel :